listener.go

```diff
@@ -74,6 +75,8 @@ func ListenersWithGroup(g Group, ls []*Listener) []*Listener {
 // TODO: add rate limiting
 func (l *Listener) accept() {
 	defer l.teardown()
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
I'd comment here and say something like 'defers are calling in LIFO order, this will execute before `l.teardown()`'